app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -161,7 +165,13 @@ class RegisterController @Inject() (val messagesApi:           MessagesApi,
       checkIfDoneEligibilityChecks { eligibleWithEmail ⇒
         eligibleWithEmail.confirmedEmail.fold[Future[Result]](
           SeeOther(routes.RegisterController.getSelectEmailPage().url))(
-            _ ⇒ Ok(views.html.register.create_account_help_to_save()))
+            _ ⇒
+              EligibilityReason.fromEligible(eligibleWithEmail.eligible).fold{
+                logger.warn(s"Could not parse eligiblity reason: ${eligibleWithEmail.eligible}", eligibleWithEmail.userInfo.nino)
+                internalServerError()
+              }{ reason ⇒
+                Ok(views.html.register.create_account_help_to_save(reason))
```

Comment:
why do we need to pass the `reason` here ?

Reply:
there's some content changes on this page which needs to be dynamic based on the eligibility reason - it's not being used right now but it's getting it ready to be used here